### PR TITLE
Db1

### PR DESCRIPTION
**Pull request by smith4312**
DB files are cleaned up and consolidated to 3 files
Controller API has stubbed out methods for completion

### CODE REVIEW

File: app/controllers/Application.scala [hunk 1]
```diff
@@ -1,12 +1,194 @@
...
+      }
+    } finally {
+      conn.close()
+    }
+    val j = JsArray(jsonBuffer)
+    Ok(j)
+  }
+
+
+  def findAccount = Action {
```
Comment (nafg) [inline app/controllers/Application.scala]: For placeholder actions, you can use `TODO`: https://playframework.com/documentation/2.4.x/ScalaActions#TODO-dummy-page

File: conf/application.conf [hunk 2]
```diff
@@ -34,6 +34,10 @@ play.i18n.langs = [ "en" ]
 # db.default.url="jdbc:h2:mem:play"
 # db.default.user=sa
 # db.default.password=""
+db.default.driver=org.postgresql.Driver
```
Comment (nafg) [inline conf/application.conf:37]: This should not be committed to git. Please rebase it out.

Comment (nafg) [inline conf/application.conf:37]: Especially not the username and password.

File: database/TAG1/USER_FUNCTIONS.func.sql [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+					country varchar(20),
+					email varchar(60),
+					phone  varchar(20),
+					mobile_phone varchar(20),
+					notes varchar(255),
+					office_location_id integer ) RETURNS text AS $$
+BEGIN
+	--Can't create a duplicate	
+	IF userName = '' THEN
+		RETURN 'User name must be at least 4 characters';
```
Comment (nafg) [inline database/TAG1/USER_FUNCTIONS.func.sql]: I'm personally not a big fan of putting logic in the database

File: database/TAG1/POPULATE_DATA.dat.sql [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+INSERT INTO TICKET_ACTION VALUES (DEFAULT, 'Remove Video', 		'Protection', 'Install a new filter on a device', '', '');
+INSERT INTO TICKET_ACTION VALUES (DEFAULT, 'Remove Wifi and Video', 'Protection', 'Install a new filter on a device', '', '');
+
+
+/*******************************************************************************
+ *
+ *	AGENT
+ *
+ *******************************************************************************/
+INSERT INTO AGENT VALUES (DEFAULT, 'David', 'Smith', 'DSMITH', 'Password', '123 Main. St.', 'Lakewood', 'NJ', '08701', 'USA',
```
Comment (nafg) [inline database/TAG1/POPULATE_DATA.dat.sql:57]: Is all of this data for testing purposes only?

File: database/TAG1/CREATE_TABLES.tbl.sql [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+	PROVIDER			VARCHAR(120),
+	URL				VARCHAR(120),
+	NOTES				VARCHAR(255)
+);
+
+
+CREATE TABLE PERSON_DEVICE
+(
+	ID				SERIAL PRIMARY KEY,
+	PERSON_ID			INT REFERENCES PERSON ON DELETE CASCADE,
```
Comment (nafg) [inline database/TAG1/CREATE_TABLES.tbl.sql:109]: Wow, I didn't realize postgres let you write it so tersely

File: app/controllers/Application.scala [hunk 6]
```diff
@@ -1,12 +1,194 @@
...
+
+    Ok
+  }
+
+
+  //TODO we want to take some parameters for this call
+  // - Office location
+  // - Agent ID for tickets assigned to that agent
+  // - can these be
+  def listTickets = Action {
```
Comment (nafg) [inline app/controllers/Application.scala:55]: I think at some point we need to factor out the redundancy between all these controller actions (and get rid of `var`s for bonus points), but I'm not sure if it should wait... (depends on the עולם's readiness...)

File: database/TAG1/USER_FUNCTIONS.func.sql (hunk 3 above)
Comment (smith4312) [inline database/TAG1/USER_FUNCTIONS.func.sql]: Yea. This was meant as error checking, not application logic. But I will switch it to a table constraint on username.

File: database/TAG1/POPULATE_DATA.dat.sql (hunk 4 above)
Comment (smith4312) [inline database/TAG1/POPULATE_DATA.dat.sql:57]: Data is for testing. I hope some of the action and status types are correct.

File: app/controllers/Application.scala (hunk 6 above)
Comment (smith4312) [inline app/controllers/Application.scala:55]: I’ve made the changes. Maybe rebase we do after the merge to get master as well. Please review again.

**Comment by nafg** [comment]
LGTM

**Pull request merged**